tests/fixture/tmpnet/start_collectors.go

```diff
@@ -0,0 +1,242 @@
+`, workingDir, username, password, workingDir)
+		},
+	)
+}
+
+func ensureCollectorRunning(
+	log logging.Logger,
+	cmdName string,
+	args string,
+	baseEnvName string,
+	configGenerator configGeneratorFunc,
+) error {
+	tmpnetDir, err := getTmpnetPath()
+	if err != nil {
+		return err
+	}
+	workingDir := filepath.Join(tmpnetDir, cmdName)
+	pidFilename := "run.pid"
+	pidPath := filepath.Join(workingDir, pidFilename)
+
+	if err := os.MkdirAll(workingDir, perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create %s dir: %w", cmdName, err)
+	}
+
+	if err := os.MkdirAll(filepath.Join(workingDir, "file_sd_configs"), perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create promtail file_sd_configs dir: %w", err)
+	}
+
+	// Read the PID from the file
+	pidData, err := os.ReadFile(pidPath)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("failed to read %s PID file %s: %w", cmdName, pidPath, err)
+	}
+	if len(pidData) > 0 {
+		pid, err := strconv.Atoi(string(pidData))
+		if err != nil {
+			return fmt.Errorf("failed to parse %s PID: %w", cmdName, err)
+		}
+		process, err := getProcess(pid)
+		if err != nil {
+			return err
+		}
+		if process != nil {
+			log.Info(cmdName + " is already running")
+			return nil
+		}
+	}
+
+	// Remove the pid file to avoid conflicting with the new one starting
+	if err := os.Remove(pidPath); err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			return fmt.Errorf("failed to remove stale pid file: %w", err)
+		}
+	} else {
+		log.Info("deleted stale "+cmdName+" pid file",
+			zap.String("path", pidPath),
+		)
+	}
+
+	// TODO(marun) Maybe collect errors instead of returning them 1-by-1?
+	if _, err := exec.LookPath(cmdName); err != nil {
+		return fmt.Errorf("%s command not found. Maybe run 'nix develop'?", cmdName)
+	}
+
+	usernameEnvVar := baseEnvName + "_USERNAME"
+	username := getEnv(usernameEnvVar, "")
+	if len(username) == 0 {
+		return fmt.Errorf("%s env var not set", usernameEnvVar)
+	}
+
+	passwordEnvVar := baseEnvName + "_PASSWORD"
+	password := getEnv(passwordEnvVar, "")
+	if len(password) == 0 {
+		return fmt.Errorf("%s var not set", passwordEnvVar)
+	}
+
+	confFilename := cmdName + ".yaml"
+	confPath := filepath.Join(workingDir, confFilename)
+	log.Info("writing "+cmdName+" config",
+		zap.String("path", confPath),
+	)
+	config := configGenerator(workingDir, username, password)
+	if err := os.WriteFile(confPath, []byte(config), perms.ReadWrite); err != nil {
+		return err
+	}
+
+	fullCmd := "nohup " + cmdName + " " + args + " > " + cmdName + ".log 2>&1 & echo -n \"$!\" > " + pidFilename
+	log.Info("starting "+cmdName,
+		zap.String("workingDir", workingDir),
+		zap.String("fullCmd", fullCmd),
+	)
+
+	// TODO(marun) Figure out a way to redirect stdout and stderr of a detached child process without a bash shell
+	cmd := exec.Command("bash", "-c", fullCmd)
+	configureDetachedProcess(cmd) // Ensure the child process will outlive its parent
```

Comment:
I believe you can create the command and set `Stdout` and `Stderr` to nil directly (according to the docs)

Reply:
I may be wrong here, just remembered reading the documentation around this, so thought I'd provide the suggestion. May not behave as I expected from reading the docs (either due to a misread or incorrect docs).

Reply:
I think that I thought you were trying to ignore the output here: https://github.com/ava-labs/avalanchego/blob/15c496b09f92cc5ac23b5aa2937d17a258d9a14f/tests/fixture/tmpnet/start_collectors.go#L182, but you actually wanted to write it to the specified file. I think you could still do this by setting `Stdout` and `Stderr` to the corresponding file and avoid constructing a bash command.